core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -138,18 +138,25 @@ os_mmap(void *hint, size_t size, int prot, int flags, os_file_handle file)
 
     /* memory hasn't been mapped or was mapped failed previously */
     if (addr == MAP_FAILED) {
-        /* try 5 times */
-        for (i = 0; i < 5; i++) {
+        /* try 5 times on EAGAIN, keep retrying on EINTR */
+        i = 0;
+        while (i < 5) {
             addr = mmap(hint, request_size, map_prot, map_flags, file, 0);
             if (addr != MAP_FAILED)
                 break;
+            if (errno == EINTR)
+                continue;
+            if (errno != EAGAIN) {
```

Comment:
Could you also retry on `ENOMEM` and update the comment? According to the discussion, we had better also retry on it: `if (error != EAGAIN && error != NOMEM)`.

Reply:
Done, thank you